core/src/main/java/org/javarosa/core/model/actions/Action.java

```diff
@@ -37,9 +46,7 @@ public Action(String name) {
      * @return TreeReference targeted by the action or null if the action
      * wasn't completed.
      */
-    public TreeReference processAction(FormDef model, TreeReference context) {
-        return null;
-    }
+    public abstract TreeReference processAction(FormDef model, TreeReference context);
```

Comment:
I tried to make `processAction` abstract before and it broke something in the J2ME build (https://github.com/dimagi/javarosa/pull/186). @ctsims  do you know what was going on?